registration/round.go

```diff
@@ -199,7 +200,19 @@ func (r *round) flushPendingSubmits() {
 	r.flushPendingSubmitsLocked()
 }
 
+func (r *round) timedFlushPendingSubmits() {
+	r.batchMutex.Lock()
+	defer r.batchMutex.Unlock()
+	if r.pendingFlush != nil {
+		r.flushPendingSubmitsLocked()
+	}
```

Comment:
This if is now always true I think?

Reply:
No, it's not. It won't be true when the timer fires while another goroutine is in `flushPendingSubmitsLocked()`. Then, `timedFlushPendingSubmits()` waits on `r.batchMutex.Lock()`, while the other goroutine clears the `r.pendingFlush`.